embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/execution/HttpUrlConnectionRequestExecutionService.kt

```diff
@@ -1,73 +1,62 @@
 package io.embrace.android.embracesdk.internal.delivery.execution
 
 import io.embrace.android.embracesdk.internal.comms.api.ApiRequestV2
-import io.embrace.android.embracesdk.internal.comms.api.ApiResponse
 import io.embrace.android.embracesdk.internal.comms.api.Endpoint
 import io.embrace.android.embracesdk.internal.delivery.SupportedEnvelopeType
+import io.embrace.android.embracesdk.internal.delivery.execution.ExecutionResult.Companion.getResult
 import java.io.InputStream
 import java.net.HttpURLConnection
 import java.net.URL
 
-private const val NO_HTTP_RESPONSE = -1
-private const val TOO_MANY_REQUESTS = 429
 private const val DEFAULT_TIMEOUT_MILLISECONDS = 10 * 1000
 
 class HttpUrlConnectionRequestExecutionService(
     private val coreBaseUrl: String,
     private val lazyDeviceId: Lazy<String>,
     private val appId: String,
     private val embraceVersionName: String,
-    private val connectionTimeoutMilliseconds: Int = DEFAULT_TIMEOUT_MILLISECONDS
+    private val connectionTimeoutMilliseconds: Int = DEFAULT_TIMEOUT_MILLISECONDS,
 ) : RequestExecutionService {
     override fun attemptHttpRequest(
         payloadStream: () -> InputStream,
         envelopeType: SupportedEnvelopeType,
-    ): ApiResponse {
+    ): ExecutionResult {
         val apiRequest = envelopeType.endpoint.getApiRequestFromEndpoint()
-
-        try {
+        var headersProvider = { emptyMap<String, String>() }
```

Comment:
nit: we could move these closer to where they're actually used

Reply:
Oh yeah I can do that

Reply:
Oh never mind - I need to use this line 45, which is why it is a var with a default value.

But looking at the code, creating that object is wasteful because we almost always replace it, so let me set it to null and do an elvis to put it in if it doesn't exist